fxa/plugins/requests.py

```diff
@@ -92,18 +125,29 @@ def __init__(self, email, password, scopes=None, client_id=None,
         self.account_server_url = account_server_url
         self.oauth_server_url = oauth_server_url
 
-    def __call__(self, request):
-        client = core.Client(server_url=self.account_server_url)
-        session = client.login(self.email, self.password, keys=True)
-
-        url = urlparse(self.oauth_server_url)
-        audience = "%s://%s/" % (url.scheme, url.netloc)
+        self.cache = cache
+        if self.cache is True:
+            self.cache = MemoryCache(ttl)
 
-        bid_assertion = session.get_identity_assertion(audience)
-        oauth_client = oauth.Client(server_url=self.oauth_server_url)
-        token = oauth_client.authorize_token(bid_assertion,
-                                             ' '.join(self.scopes),
-                                             self.client_id)
+    def __call__(self, request):
+        cache_key = get_cache_key(
+            self.account_server_url, self.oauth_server_url,
+            self.email, self.password, self.scopes, self.client_id)
+        token = self.cache.get(cache_key)
```

Comment:
Otherwise test here if we can use the cache otherwise just bypass it (if cache is set to False)